che-mount/entrypoint.sh

```diff
@@ -93,45 +107,95 @@ trap 'stop_sync' SIGHUP SIGTERM SIGINT
 init_logging
 init_global_variables
 parse_command_line "$@"
-status=$?
-if [ $status -ne 0 ]; then
-    exit 1
+[ "$(ls -A /mnthost )" ] && EMPTY=false || EMPTY=true
+if [ $EMPTY = false ]; then
+    warn "(che mount): Local folder is not empty. Are you sure?[Y/n]"
+    read yn
+    case $yn in
+        [Yy]*) 
+            break
+            ;;
+        [Nn]*) 
+            exit
+            ;;
+        *) 
+            info "(che mount): Please answer yes or no."
+            ;;
+    esac
 fi
-
 docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock eclipse/che-action:${CHE_VERSION} get-ssh-data ${WORKSPACE_NAME} ${COMMAND_EXTRA_ARGS} > $HOME/env
 if [ $? -ne 0 ]; then
     error "ERROR: Error when trying to get workspace data for workspace named ${WORKSPACE_NAME}"
     echo "List of workspaces are:"
-    docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock eclipse/che-action:${CHE_VERSION} list-workspaces
+    docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock eclipse/che-action:${CHE_VERSION} list-workspaces ${COMMAND_EXTRA_ARGS} --sync-agent
     return 1
 fi
-
 source $HOME/env
 
 # store private key
-mkdir $HOME/.ssh
+mkdir -p $HOME/.ssh
```

Comment:
it's still incorrect as agents are per machine. So you can't say sync agent is enabled for a workspace. It's only in the case of the dev-machine.
So you should either list agent for each machine of the workspace or not list it at all